cmd/acr/purge.go

```diff
@@ -6,25 +6,144 @@ package main
 import (
 	"fmt"
 	"io"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
 
-	acrapi "github.com/Azure/libacr/golang"
+	"github.com/AzureCR/acr-cli/cmd/api"
 	"github.com/spf13/cobra"
 )
 
 const (
-	purgeLongMessage = `` // TODO
+	purgeLongMessage = `Purge the registry, given the registry name and a repository name this 
+command untags all the tags that match with the filter and that are older 
+than a duration, after that, all manifests that do not have any tags 
+associated with them also get deleted.`
 )
 
+var registryName string
+var username string
+var password string
+var ago string
+var dangling bool
+var filter string
+var repoName string
+
 func newPurgeCmd(out io.Writer) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "purge",
-		Short: "", // TODO
+		Short: "Delete images from a registry.",
 		Long:  purgeLongMessage,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			fmt.Println("The default base uri is:", acrapi.DefaultBaseURI)
+			var wg sync.WaitGroup
+			loginURL := api.LoginURL(registryName)
+			auth := api.BasicAuth(username, password)
+			if !dangling {
+				var days int
+				var durationString string
+				if strings.Contains(ago, "d") {
+					if _, e := fmt.Sscanf(ago, "%dd%s", &days, &durationString); e != nil {
```

Comment:
Because go format does not support days I try to parse something like an extended go duration, if there is not a day specified then I assume it to be a regular duration string and I try to parse, also the days are set to 0 because they were not specified.